src/character_guns.cpp

```diff
@@ -164,8 +164,9 @@ std::pair<int, int> Character::gunmod_installation_odds( const item_location &gu
     return std::make_pair( roll, risk );
 }
 
-void Character::gunmod_add( item &gun, item &mod )
+void Character::gunmod_add( item_location gun_loc, item &mod )
 {
+    item gun = *gun_loc.get_item();
```

Comment:
```suggestion
    item &gun = *gun_loc;
```

This shouldn't create a copy of the item.